lib/train/transports/ssh.rb

```diff
@@ -76,7 +76,7 @@ def connection(state = {}, &block)
       validate_options(opts)
       conn_opts = connection_options(opts)
 
-      if defined?(@connection) && @connection_options == conn_opts
+      if reusable_connection?(conn_opts)
```

Comment:
Why remove the `defined?(@connection)` test? I don't foresee the case where `@connection` would be `nil` but it seems wise to guard against the possibility of it.

Reply:
Good point. Overambitious optimization on my side :)